packages/shared/lib/core/wallet/stores/selected-account-activities.store.ts

```diff
@@ -101,26 +99,59 @@ export const queriedActivities: Readable<Activity[]> = derived(
 
         activityList = activityList.filter((activity) => isVisibleActivity(activity))
 
-        if (activitySearchTerm) {
-            activityList = activityList.filter(
-                (activity) =>
-                    (activity.data.type === ActivityType.Transaction &&
-                        ((activity.data.recipient?.type === 'account' &&
-                            activity.data.recipient?.account?.name === $activitySearchTerm) ||
-                            (activity.data.recipient?.type === 'address' &&
-                                activity.data.recipient?.address === $activitySearchTerm))) ||
-                    activity?.id?.toLowerCase() === $activitySearchTerm ||
-                    ($activitySearchTerm[0] === '>' &&
-                        unitToValue($activitySearchTerm.substring(1)) < activity.data.rawAmount) ||
-                    ($activitySearchTerm[0] === '<' &&
-                        unitToValue($activitySearchTerm.substring(1)) > activity.data.rawAmount) ||
-                    ($activitySearchTerm[1] === 'i' &&
-                        isValueInUnitRange(activity.data.rawAmount, $activitySearchTerm)) ||
-                    activity.data.rawAmount === unitToValue($activitySearchTerm) ||
-                    formatUnitBestMatch(activity.data.rawAmount).toString().toLowerCase()?.includes($activitySearchTerm)
-            )
+        if ($activitySearchTerm) {
+            activityList = activityList.filter((activity) => {
+                const fieldsToSearch = getFieldsToSearchFromActivity(activity)
+                return fieldsToSearch?.find((field) =>
+                    field?.toLowerCase()?.includes($activitySearchTerm?.toLowerCase())
+                )
+            })
         }
 
         return activityList.sort((activity1, activity2) => activity2.time.getTime() - activity1.time.getTime())
     }
 )
+
+function getFieldsToSearchFromActivity(activity: IActivity): string[] {
+    const fieldsToSearch: string[] = []
+
+    if (activity?.transactionId) {
```

Comment:
you can remove all of the `?`  because activity is never nullish here (as well as activity.data)